### ISSUE TITLE
Initial search fails with "No internet connection" error, in the second step modes of transport are missing (app)

### ISSUE THREAD

**Issue by stadtnavimanager** [stadtnavi/stadtnavi_app#390]
**Description:**

In the test version, the first search query often results in an error message stating "**Exception: Internet no connection**," even though a stable internet connection is present. 
When the search is restarted, the i**ntermodal transport options at the top are missing**. Only on the third attempt do the search results display correctly.

---

**Steps to reproduce:**
--> note: this does not happen everytime! It seems to happen more often, if the app was not used for some time. e.g. in the morning. But my intempts to always reproduce it do fail (also deleting the cache and reinstalling the app=

1. Open the test version of the application.
2. Perform an initial search query.
3. Observe the "No internet connection" error message.
4. Restart the search and note the missing intermodal transport options.
5. Perform a third search, where the results finally appear as expected.

---

**Observed behavior:**

- First search: Error message stating "No internet connection," despite a working internet connection.
 ![Image](https://github.com/user-attachments/assets/8a95e44a-9b67-4c58-9bda-b99cfa9d3669)

- Second search: Results load, but intermodal transport options are missing.

![Image](https://github.com/user-attachments/assets/11978a81-b9f3-49cb-a460-fec1d9642c79)

- Third search: Results display correctly, including intermodal transport options.
![Image](https://github.com/user-attachments/assets/d74d03ff-749a-402a-946d-bbb68c0e2cfa)

**Expected behavior:**

The first search query should load results correctly without errors or missing intermodal transport options.

---

**Environment:**

- **Version:** 1.12.0
- **Operating System:** Android 14, Google Pixel 7a

---

**Additional information:**

This issue significantly disrupts the user experience and may discourage users from continuing to use the application. A fix to ensure reliable performance on the first search attempt is critical.

**Comment by leonardehrenfried** [comment]
I find the dev backend pretty fast (web): https://dev.stadtnavi.eu/reiseplan/Stuttgart%3A%3A48.7784485%2C9.1800132/Ehningen%3A%3A48.6592556%2C8.9405097?time=1733921835

Is it for you the case as well?

**Comment by stadtnavimanager** [comment]
For me it depends what we are talking about.
The routing is pretty normal/ fast.

Loading elements on the map however is pretty slow, e.g. if I move the map or if I add more map data in the filters (until the icons are shown, appear in the right size etc.).

Does this answer the question?

**Comment by stadtnavimanager** [comment]
Also, when something on the map is still loading, I cannot really enter a new destination in the routing

**Comment by stadtnavimanager** [comment]
For the documentation: I experienced this problem again today

**Comment by stadtnavimanager** [comment]
For the documentation:
yesterday evening again the same problem.
then, after loading the route again, only RegioRad suggestions were shown, even though all modes of transport were selected:
![image](https://github.com/user-attachments/assets/4593bd61-4d4b-4eef-be27-a2d2e18466b7)
![image](https://github.com/user-attachments/assets/8cbd7a1d-d73f-4a45-be49-477164e972db)

**Comment by leonardehrenfried** [comment]
So it turns out there _was_ a performance regression for the OTP startup which was taking a lot longer than usual. This was because of on an unrelated feature and very much unintentional. It has been reverted and we are working on a long-term solution upstream so this doesn't happen again.

In the meantime I deployed a fix so this should no longer happen.

@stadtnavimanager Have you been seeing this in the last week?

**Comment by stadtnavimanager** [comment]
Great, thanks for the update and the fix @leonardehrenfried 

I am only back in office since this morning and did not test last week, so I did not observe anything lately

**Comment by Denis306** [comment]
I will move this to "In Review" so we know we have to retest  Leonards changes first before doing anything else.

**Comment by stadtnavimanager** [comment]
Unfortunately the issue occured again yesterday morning when I first opened the app.
First time I open: "error connection"

![Image](https://github.com/user-attachments/assets/e03247d1-7e65-45f9-aa8a-ce10d320bd84)

Second time I open: missing modes of transport

![Image](https://github.com/user-attachments/assets/f55277c3-65ba-4ef6-adf3-d0e3b14c9afb)

It worked on the third try

**Comment by leonardehrenfried** [comment]
Hmm, that's disappointing. I have now configured a warm up request. Lets see if that helps.

@Denis306 Do you have an idea what the missing modes could be due to?

**Comment by Denis306** [comment]
Can be tested iOS and Androdi V1.14.1. 
@hbruch @stadtnavimanager I created a new ticket for the error handling. Samuel will take a look and optimize it.

**Comment by stadtnavimanager** [comment]
for me it looks good o far. @hbruch could you also test on your phone and close the ticket, if the probem does not occur anymore? thx
@Denis306 could you link the ticket you opened? thx

**Comment by stadtnavimanager** [comment]
Unfortunately I have to reopen this issue @Denis306 @SamuelRioTz 

Yesterday there were also problems with the correct time display, even though I selected a time several times. in the end the ols known error was shown again.

<img width="204" alt="Image" src="https://github.com/user-attachments/assets/dfd14ff0-a9d2-4c1a-adcb-bd4d14e8a8c5" />

<img width="197" alt="Image" src="https://github.com/user-attachments/assets/d8896aa2-d2fe-4186-ab21-a919f819083e" />

<img width="191" alt="Image" src="https://github.com/user-attachments/assets/8c2b5a19-993c-4c69-9824-c7dec80e6b5f" />

This morning it is still very slow, sometimes it shows a route, but without the intermodal routing options, sometimes it shows the known error:

<img width="234" alt="Image" src="https://github.com/user-attachments/assets/15dd27c8-ec05-4cb0-8841-c6479b8e465f" />

<img width="191" alt="Image" src="https://github.com/user-attachments/assets/d6196bfc-952f-4f84-9b2d-41bbb1b0a4ce" />

Version 1.14.5

**Comment by stadtnavimanager** [comment]
I saw that there was another update yesterday evening. I updated now and tested again.

Unfortunately, the performance issues are still there, the route is not loading completely:

1st try

<img width="191" alt="Image" src="https://github.com/user-attachments/assets/22f640ca-c11a-4442-9863-fdf7b77933c7" />

2nd try

<img width="208" alt="Image" src="https://github.com/user-attachments/assets/53cdbd66-795c-49a4-8dbb-b3635402ecfa" />

3rd try

<img width="194" alt="Image" src="https://github.com/user-attachments/assets/9d60cbbf-1f72-4fb0-9793-e5590b8866c0" />

4th try

<img width="192" alt="Image" src="https://github.com/user-attachments/assets/74f4dc6a-d97c-497c-b44c-f0309ba55206" />

Version 1.14.6 (Android)

**Comment by stadtnavimanager** [comment]
Unfortunately, the problem is still ocurring:

<img width="199" alt="Image" src="https://github.com/user-attachments/assets/0d6c6482-ebb5-4d20-b3f3-9b1a5d5f8220" />

Version 1.14.7

@SamuelRioTz @Denis306 we cannot launch the new version of the app and the webversion, until this issue is not solved

**Comment by Denis306** [comment]
@stadtnavimanager can you do some more tests? There is a new version.

**Comment by stadtnavimanager** [comment]
@Denis306 so far I did not get the error message yet. But it still takes around 8 seconds between the first route is shown until the intermodal routes are shown at the top, which seems to be quite long

![Image](https://github.com/user-attachments/assets/8d06a955-b4eb-4ba2-aa77-218dcbccdfd6)

**Comment by Denis306** [comment]
This can be tested on V 1.14.20 (120) in Android and iOS.

**Comment by stadtnavimanager** [comment]
Not experiencing the problems anymore. Thanks! :)
